rclpy/src/rclpy/_rclpy.c

```diff
@@ -167,7 +167,7 @@ rclpy_create_guard_condition(PyObject * Py_UNUSED(self), PyObject * args)
     return NULL;
   }
 
-  PyObject * pygc_impl_reference = PyLong_FromUnsignedLongLong((uint64_t)&gc->impl);
+  PyObject * pygc_impl_reference = PyLong_FromVoidPtr(gc);
```

Comment:
@sloretz consider
```suggestion
  PyObject * pygc_reference = PyLong_FromVoidPtr(gc);
```
instead.

Reply:
49aa8f0